### DISCUSSION TITLE
Populating test data for an existing installation

### DISCUSSION THREAD

**Discussion by michaelbromley** [vendurehq/vendure#1868] [How-to Guides]
When installing with `@vendure/create`, you have the option of populating test data (products, payment methods, countries, zones, tax rates etc).

This guide illustrates how to populate that test data again on an existing Vendure installation, without needing to re-install from scratch.

1. `npm install --save-dev @vendure/create`. This installs the "create" package, which contains the test data we will need.
2. drop all tables from your database, but leave the actual database there.
3. create a `populate.ts` script that looks like this:

```ts
import { populate } from '@vendure/core/cli';
import { bootstrap, VendureConfig } from '@vendure/core';
import { config } from './vendure-config';

populate(
    () => bootstrap({
        ...config,
        importExportOptions: {
            importAssetsDir: path.join(
                require.resolve('@vendure/create/assets/products.csv'),
                '../images'
            ),
        },
        dbConnectionOptions: {...config.dbConnectionOptions, synchronize: true}
    }),
    require('@vendure/create/assets/initial-data.json'),
    require.resolve('@vendure/create/assets/products.csv')
).then(app => app.close())
```

Running that script will re-create your database tables and then populate with fresh test data.

**Comment by zikzak-ai** [comment]
still works.  just need to add 
import * as path from "path";
to the top the file.

**Comment by sbourdeaud** [comment]
asset images are not rendering correctly after this (using postgres database)
<img width="1206" height="635" alt="image" src="https://github.com/user-attachments/assets/395836ce-8850-4d08-8a6c-1f57f0f81148" />

**Reply by sbourdeaud** [reply]
nvm it was me not setting the asset URL correctly in vendure-config.ts
